File: workspace/pset6/cash/cash.py
# Greedy algorithm

from cs50 import get_float


# Get change input
change = -1
while change < 0:
    change = get_float("Change owed: ")

# Convert change to cents
change *= 100
change = round(change)

# Set minimum number of coins
min_coins = 0

# Determine how many coins of each coin_type we'll use starting from biggest
coin_types = [25, 10, 5, 1]
for coin_value in coin_types:
    while change >= coin_value:
        change -= coin_value
        min_coins += 1

print(f"{min_coins}")

File: workspace/pset1/cash/cash.c
// Greedy algorithm

#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main()
{
    // Get how much change is owed
    float change;
    do
    {
        change = get_float("Change owed: ");
    }
    while (change < 0);

    // Convert change to cents
    change *= 100;
    int change_in_cents = roundf(change);

    // Determine minimum number of coins
    int min_coins = 0;
    int coin_value;
    for (int coin_index = 0; coin_index < 4; coin_index++)
    {
        // Choose coin value from available coins in descending order
        switch (coin_index)
        {
            case 0:
                coin_value = 25;
                break;

            case 1:
                coin_value = 10;
                break;

            case 2:
                coin_value = 5;
                break;

            case 3:
                coin_value = 1;
                break;
        }

        // Use as many coins of value: coin_value as possible
        while (change_in_cents >= coin_value)
        {
            change_in_cents -= coin_value;
            min_coins += 1;
        }
    }
    printf("%i\n", min_coins);
}